onnxruntime/python/torch_cpp_extensions/aten_op_executor/aten_op_executor.cc

```diff
@@ -6,42 +6,24 @@
 #include <unordered_map>
 #include <vector>
 
-template <typename T>
+template <typename TSrc, typename TDst>
 c10::IValue ToIValue(const DLManagedTensor* dlpack, bool is_optional) {
   TORCH_INTERNAL_ASSERT((dlpack->dl_tensor.ndim == 0 && dlpack->dl_tensor.shape == nullptr) ||
                         (dlpack->dl_tensor.ndim == 1 && dlpack->dl_tensor.shape[0] == 1));
-  T value = *reinterpret_cast<const T*>(dlpack->dl_tensor.data);
-  return is_optional ? c10::IValue(c10::optional<T>(value)) : c10::IValue(value);
+  TDst value = static_cast<TDst>(*reinterpret_cast<const TSrc*>(dlpack->dl_tensor.data));
+  return is_optional ? c10::IValue(c10::optional<TDst>(value)) : c10::IValue(value);
```

Comment:
Was `c10::IValue(value)` performing an implicit cast until now? Why is the change required now (does the implicit cast no longer work)?

Reply:
IValue(double) can take float as input, but IValue(c10::List) cannot take c10::List<float>. That's why I need this change.